### PR TITLE
feat(openapi): adapt OpenAPIv2 "Update data"

### PR DESCRIPTION
**Pull request by wangyuqing0424**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-27rqzfx OpenAPIv2 对接 —— UpdateData](https://app.clickup.com/t/27rqzfx)

File: graviti/openapi/data.py [hunk 1]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+    /sheets/{sheet}/data`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        data:  The update data.
```
Comment (linjiX) [inline graviti/openapi/data.py:204]: remove a space here

File: graviti/openapi/data.py [hunk 2]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+
+def update_data(
+    access_key: str,
+    url: str,
+    owner: str,
+    dataset: str,
+    *,
+    draft_number: int,
+    sheet: str,
+    data: List[Any],
```
Comment (linjiX) [inline graviti/openapi/data.py:190]: should the type be `List[Dict[str, Any]]`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 3]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+        ...                         "type": "remoteFile"
+        ...                     },
+        ...                     "attribute": {
+        ...                         "weather": "clear",
+        ...                         "scene": "city street",
+        ...                         "timeofday": "daytime"
+        ...                     },
+        ...                 },
+        ...                 ...(total 5 items)
+        ...             ]
```
Comment (graczhual) [inline graviti/openapi/data.py:232]: There seems to be a problem with indentation here.

File: graviti/openapi/data.py [hunk 4]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+        ... )
+
+    """
+    url = urljoin(url, f"/v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets/{sheet}/data")
```
Comment (graczhual) [inline graviti/openapi/data.py:236]: f"v2", do not use `/` here, urljoin will cause some problems

File: graviti/openapi/data.py [hunk 5]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+        ...                         "checksum": "dcc197970e607f7576d978972f6fb312911ce005",
+        ...                         "type": "remoteFile"
+        ...                     },
+        ...                     "attribute": {
+        ...                         "weather": "clear",
+        ...                         "scene": "city street",
+        ...                         "timeofday": "daytime"
+        ...                     },
+        ...                 },
+        ...                 ...(total 5 items)
```
Comment (graczhual) [inline graviti/openapi/data.py:231]: in the example, use the right code,  do not use`...(total 5 items)`. we could write two rows here

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 6]
```diff
@@ -177,3 +177,65 @@ def list_commit_data(
...
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        data: The update data.
+        offset: The starting position of the update data.
+        order_by: The string determine the order of sorting by  the order in
```
Comment (graczhual) [inline graviti/openapi/data.py:206]: one space after `sorting by`

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 7]
```diff
@@ -177,3 +177,77 @@ def list_commit_data(
...
+        ...                         "url": "https://content-store-prod-vers",
+        ...                         "checksum": "dcc197970e607f7576d978972f6fb2a2881ce004",
+        ...                         "type": "remoteFile"
+        ...                     },
+        ...                     "attribute": {
+        ...                         "weather": "clear",
+        ...                         "scene": "highway",
+        ...                         "timeofday": "dawn/dusk"
+        ...                     },
+        ...                 },...
```
Comment (graczhual) [inline graviti/openapi/data.py:243]: do not need `,...`here

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 8]
```diff
@@ -177,3 +177,77 @@ def list_commit_data(
...
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        data: The update data.
+        offset: The starting position of the update data.
+        order_by: The string determine the order of sorting by the order in
+            which column names appear.
```
Comment (graczhual) [inline graviti/openapi/data.py:207]: add the description about None.

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 9]
```diff
@@ -55,8 +55,9 @@ def list_draft_data(
         sheet: The sheet name.
         columns: The string of column names separated by commas.
```
Comment (graczhual) [inline graviti/openapi/data.py:56]: comma -> `|`

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 10]
```diff
@@ -177,3 +179,79 @@ def list_commit_data(
...
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        data: The update data.
+        offset: The starting position of the update data.
+        order_by: The string determine the order of sorting by the order in which column names
+            appear. Column names are seperated by ``|``. If None or incomplete, the remainder
```
Comment (graczhual) [inline graviti/openapi/data.py:209]: Multiple indexes can be expressed using dots.

**Review by graczhual — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

File: graviti/openapi/data.py [hunk 11]
```diff
@@ -177,3 +179,79 @@ def list_commit_data(
...
+        ...     "OxfordIIITPet",
+        ...     draft_number = 1,
+        ...     sheet = "train",
+        ...     data = [
+        ...                 {
+        ...                     "filename": "0000f77c-6257be58.jpg",
+        ...                     "image": {
+        ...                         "url": "https://content-store-prod-vers",
+        ...                         "checksum": "dcc197970e607f7576d978972f6fb312911ce005",
+        ...                         "type": "remoteFile"
```
Comment (linjiX) [inline graviti/openapi/data.py:226]: why we need the `"type"` here?

File: graviti/openapi/data.py [hunk 12]
```diff
@@ -177,3 +179,79 @@ def list_commit_data(
...
+        ...                     },
+        ...                     "attribute": {
+        ...                         "weather": "clear",
+        ...                         "scene": "highway",
+        ...                         "timeofday": "dawn/dusk"
+        ...                     },
+        ...                 }
+        ...            ],
+        ...     order_by = "filename|attribute.weather",
```
Comment (linjiX) [inline graviti/openapi/data.py:248]: shall we add `offset` to the example?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**